test/shared/axiosHttpClientTests.ts

```diff
@@ -117,6 +117,16 @@ describe("axiosHttpClient", () => {
     }
   });
 
+  it("should not overwrite a user-provided cookie", async function() {
+    const client = new AxiosHttpClient();
+    const request1 = new WebResource(`${baseURL}/set-cookie`);
+    await client.sendRequest(request1);
+
+    const request2 = new WebResource(`${baseURL}/cookie`, "GET", undefined, undefined, { Cookie: "data=abcdefg" });
```

Comment:
Maybe add another `/cookie` request here without a Cookie header that demonstrates that the "Cookie": "data=123456" cookie is in the response headers. That would contrast the current second request where the "Cookie": "data=abcdefg" header overrides the existing cookie in the cookieJar.